includes/platform-checkout/class-platform-checkout-order-success-page.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Class Platform_Checkout_Order_Success_Page
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Platform_Checkout;
+
+defined( 'ABSPATH' ) || exit;
+
+use Exception;
+use WC_Order;
+use WC_Payment_Gateway_WCPay;
+use WC_Payments;
+use WC_Payments_Account;
+use WC_Payments_Action_Scheduler_Service;
+use WC_Payments_API_Client;
+use WC_Payments_Order_Service;
+use WC_Payments_Utils;
+use WCPay\Exceptions\Process_Payment_Exception;
+use WCPay\Logger;
+
+/**
+ * Handling for when customers come back to the merchant store after platform checkout.
+ */
+class Platform_Checkout_Order_Success_Page {
+	/**
+	 * Internal ID of the payment gateway.
+	 *
+	 * @type string
+	 */
+	const GATEWAY_ID = 'woopay';
+
+	/**
+	 * WCPay order service instance.
+	 *
+	 * @var WC_Payments_Order_Service
+	 */
+	private $order_service;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WC_Payments_Account                  $account The merchant account object.
+	 * @param WC_Payments_API_Client               $api_client The payments api client object.
+	 * @param WC_Payments_Order_Service            $order_service The order service object.
+	 * @param WC_Payments_Action_Scheduler_Service $action_scheduler_service The action scheduler service object.
+	 */
+	public function __construct(
+		WC_Payments_Account $account,
+		WC_Payments_API_Client $api_client,
+		WC_Payments_Order_Service $order_service,
+		WC_Payments_Action_Scheduler_Service $action_scheduler_service
+	) {
+		$this->account                  = $account;
+		$this->order_service            = $order_service;
+		$this->payments_api_client      = $api_client;
+		$this->action_scheduler_service = $action_scheduler_service;
+
+		add_filter( 'wp', [ $this, 'maybe_process_platform_checkout_redirect' ] );
+	}
+
+	/**
+	 * Doc comment.
+	 *
+	 * @return void
+	 */
+	public function maybe_process_platform_checkout_redirect() {
+		if ( ! is_order_received_page() ) {
+			return;
+		}
+
+		$payment_method = isset( $_GET['wc_payment_method'] ) ? wc_clean( wp_unslash( $_GET['wc_payment_method'] ) ) : '';
+		if ( self::GATEWAY_ID !== $payment_method ) {
+			return;
+		}
+
+		$from_woopay = isset( $_GET['from_woopay'] ) ? filter_var( wc_clean( wp_unslash( $_GET['from_woopay'] ) ), FILTER_VALIDATE_BOOL ) : false;
+		if ( ! $from_woopay ) {
+			return;
+		}
+
+		$order_id  = isset( $_GET['order_id'] ) ? wc_clean( wp_unslash( $_GET['order_id'] ) ) : '';
+		$last_4    = isset( $_GET['last4'] ) ? wc_clean( wp_unslash( $_GET['last4'] ) ) : '';
+		$intent_id = isset( $_GET['payment_intent'] ) ? wc_clean( wp_unslash( $_GET['payment_intent'] ) ) : '';
```

Comment:
These accesses to the `$_GET` variable all throw a linting error because we're not validating a nonce. Since the redirect should be coming from WooPay I'm not even sure if there is a nonce to validate so I'm not really sure how to handle that at the moment.

Maybe we need to send a nonce with the WooPay session init request and include it on the redirect for verification?

Reply:
It looks like passing a nonce through the WooPay process is the way to go (see https://github.com/Automattic/woocommerce-payments/pull/4604/commits/70466f1609e95c900a4624c35af374e62a0394da).